pages/assets/[id].tsx

```diff
@@ -157,55 +155,52 @@ export const getServerSideProps: GetServerSideProps = async ({ req, query }) =>
     }
   }
 
-  const makeCollibraServiceRequest = makeCollibraService({ authorization: `Bearer ${token.accessToken}` })
-
   try {
-    const asset = await makeCollibraServiceRequest(getAssetByID)(id)
-
-    if (!asset || !asset.approved || !Asset.isDataProduct(asset)) {
-      return {
-        notFound: true,
+    // the adapter service has a bug where it sends two requests.
+    // the first request goes through, while the second fails.
+    // TODO: refactor services
+    const { data: asset } = await axios.get<Asset>(`${config.ADAPTER_SERVICE_API_URL}/assets/${id}`, {
+      headers: {
+        authorization: `Bearer ${token.accessToken}`,
+      },
+      params: {
+        code: config.ADAPTER_SERVICE_APP_KEY,
+      },
+    })
+
+    const { data: maintainers } = await axios.get<Maintainer[]>(
+      `${config.ADAPTER_SERVICE_API_URL}/assets/${id}/maintainers`,
+      {
+        headers: {
+          authorization: `Bearer ${token.accessToken}`,
+        },
+        params: {
+          code: config.ADAPTER_SERVICE_APP_KEY,
+        },
       }
-    }
-
-    const attributes = await makeCollibraServiceRequest(getAssetAttributes)(id, "description", "timeliness")
-    const description = attributes.find((attr) => attr.type.name.toLowerCase() === "description")?.value ?? null
-    const updateFrequency = attributes.find((attr) => attr.type.name.toLowerCase() === "timeliness")?.value ?? null
-
-    asset.description = getPortableText(description)
-
-    asset.updateFrequency = getPortableText(updateFrequency)
-
-    let responsibilities = await makeCollibraServiceRequest(getAssetResponsibilities)(id)
-
-    if (!responsibilities) {
-      console.warn("[AssetDetailView] No responsibilites found for", id)
-    }
-
-    // filter out results that are not a "User" type
-    responsibilities = responsibilities.filter((responsibility) => responsibility.owner.type === "User")
-
-    const users = await Promise.all(
-      responsibilities.map((responsibility) => makeCollibraServiceRequest(getUser)(responsibility.owner.id))
     )
 
-    const responsibilitiesData: ResponsibilitiesContentSections = responsibilities.reduce((data, responsibility) => {
-      const user = users.find((u) => u.id === responsibility.owner.id)
-      if (!user) return data
-
-      const responsibilityName = responsibility.name.toUpperCase().replace(/\s/g, "_")
-
-      return {
-        ...data,
-        [responsibilityName]: [...(data[responsibilityName] ?? []), user],
-      }
-    }, {} as ResponsibilitiesContentSections)
-
     return {
       props: {
-        asset: JSON.parse(JSON.stringify(asset)),
+        asset,
         collibraBaseUrl: process.env.COLLIBRA_BASE_URL || "",
-        responsibilitiesData: JSON.parse(JSON.stringify(responsibilitiesData)),
+        responsibilitiesData: Array.isArray(maintainers)
```

Comment:
I find this one a little bit hard to read tbh :sweat_smile: Can we at least move the data manipulation outside of the actual return statement to make it easier to see the actual return object of getServerSideProps?